federation/cluster/common.sh

```diff
@@ -267,58 +266,68 @@ function create-federation-apiserver-certs {
 
 # Required
 # FEDERATION_PUSH_REPO_BASE: the docker repo where federated images will be pushed
-
-# Optional
-# FEDERATION_IMAGE_TAG: push all federated images with this tag. Used for ci testing
+# FEDERATION_IMAGE_TAG: the tag of the image to be pushed
 function push-federation-images {
     : "${FEDERATION_PUSH_REPO_BASE?Must set FEDERATION_PUSH_REPO_BASE env var}"
-    local FEDERATION_BINARIES=${FEDERATION_BINARIES:-"federation-apiserver federation-controller-manager"}
+    : "${FEDERATION_IMAGE_TAG?Must set FEDERATION_IMAGE_TAG env var}"
 
-    local imageFolder="${KUBE_ROOT}/_output/${KUBE_BUILD_STAGE}/server/${KUBE_PLATFORM}-${KUBE_ARCH}/kubernetes/server/bin"
+    source "${KUBE_ROOT}/build/common.sh"
+    source "${KUBE_ROOT}/hack/lib/util.sh"
 
-    if [[ ! -d "$imageFolder" ]];then
-	echo "${imageFolder} does not exist! Run make quick-release or make release"
-	exit 1
+    local FEDERATION_BINARIES=${FEDERATION_BINARIES:-"hyperkube"}
+
+    local bin_dir="${KUBE_ROOT}/_output/${KUBE_BUILD_STAGE}/server/${KUBE_PLATFORM}-${KUBE_ARCH}/kubernetes/server/bin"
+
+    if [[ ! -d "${bin_dir}" ]];then
+        echo "${bin_dir} does not exist! Run make quick-release or make release"
+        exit 1
     fi
 
-    for binary in $FEDERATION_BINARIES;do
-	local imageFile="${imageFolder}/${binary}.tar"
+    for binary in ${FEDERATION_BINARIES}; do
+        local bin_path="${bin_dir}/${binary}"
 
-	if [[ ! -f "$imageFile" ]];then
-	    echo "${imageFile} does not exist!"
-	    exit 1
-	fi
+        if [[ ! -f "${bin_path}" ]]; then
+            echo "${bin_path} does not exist!"
+            exit 1
+        fi
 
-	echo "Load: ${imageFile}"
-	# Load the image. Trust we know what it's called, as docker load provides no help there :(
-	docker load < "${imageFile}"
+        local docker_build_path="${bin_path}.dockerbuild"
+        local docker_file_path="${docker_build_path}/Dockerfile"
 
-	local srcImageTag="$(cat ${imageFolder}/${binary}.docker_tag)"
-	local dstImageTag="${FEDERATION_IMAGE_TAG:-$srcImageTag}"
-	local srcImageName="${FEDERATION_IMAGE_REPO_BASE}/${binary}:${srcImageTag}"
-	local dstImageName="${FEDERATION_PUSH_REPO_BASE}/${binary}:${dstImageTag}"
+        rm -rf ${docker_build_path}
```

Comment:
@luxas it doesn't make sense to add that loop here. These images are built on the fly,  only during development. It doesn't make sense to build the images for all the platforms while you are deploying for one of them. 

One thing we could do is build for $KUBE_ARCH. But I will do that in a follow up.